docs/theming.rst

```diff
@@ -130,8 +134,9 @@ The following keys are currently supported:
 
     The parent is so you don’t have to create a full theme each time: just
     create an empty theme, set the parent, and add the bits you want modified.
-    You **must** define a parent, otherwise many features won’t work due to
-    missing templates, messages, and assets.
+    It is strongly recommended you define a parent.  If you don't, many features
```

Comment:
Nope. Every theme must inherit from at least `base`/`base-jinja`, we explicitly refuse to support any other scenario (even if the code does not prevent this, we would prefer not to admit that it is a possibility, and we reserve the right to break it at any time.)

Reply:
Thinking about it a bit further: From my point of view, it would be useful if Nokia would simply officially state that

* it is possible to go without a parent template
* things might break on any Nikola version update, as new template files become required from the code.

As for support, that would mean for Nokia a new template requirement needs to be announced in the change file.

Would that be something you'd be willing to do if asked nicely?

Reply:
Here’s some example wording that might work:

> While it is possible to create a theme without a parent, it is **strongly discouraged** and not officially supported. The `base` and `base-jinja` themes provide assets, messages, and generic templates that Nikola expects to be able to use in all sites. That said, if you are making something very custom, Nikola will not prevent the creation of a theme without `base`, but you will need to manually determine which templates and messages are required in your theme.

It’s unlikely we’ll ever introduce a new template, but I guess it might be mentioned in the changelog if that happens.

Reply:
As far as the changelog is concerned: Anything in the templates can be user-changed. So I think Nikola has to mention any change in the template arena in the changelog anyway. I see that as independent of the question "with base or not".

Reply:
Some stance as the one you suggest would answer my concerns.

For some details: I would prefer to warn people of consequences, yes, but otherwise treat them as adults who are able to make grown-up decisions. So I would think it nicer if we left out the "strongly discouraged" part. Maybe warn people that growing their own theme is harder than they might think at first sight.

As for the "not officially supported": Before people file a bug, we should require them to add the pertinent parent. If that makes the bug go away, it must not be filed.

Anything else the "not officially supported" entails?

Reply:
“Strongly discouraged” and “not officially supported” are meant to suggest “don’t do this unless you really know what you’re doing”. For user-facing software, phrases like this are a good idea IMO to clearly show the limitations.

“Not officially supported” means basically what you said, i.e. we won’t help with issues caused by a parentless theme, and we don’t guarantee it will always work.

Reply:
Via the polyfill discussion, a solid reason surfaced why Nikola might rethink its policy of not officially supporting templates without parent.

Where GDPR rules, one needs a certain written agreement with any company before sending web traffic there way. This is a consequence of [GDPR art 28](https://gdpr-info.eu/art-28-gdpr/), in particular section 2.. (I've personally been one of many developers who ported all references to Google fonts to locally hosted, in the web presence of a big Germany company (some 300.000+ employed).)

I believe small, private web sites are inside the target group of Nikola. People responsible for such web sites might not have the time and legal power to establish all agreements needed. So they may choose to use no external resources, but self-host everything their web site needs.

Nikola makes not promises to that effect presently. Indeed, build-in Nikola templates freely use external resources. Any Nikola version update might conceivably introduce or change external references via the HTML produced. Wanting to play it safe is a very solid reason to use a template without parent.

Reply:
Generally there's the option `use_cdn`, but as #3763 shows this isn't always respected, like for `polyfill.io`.

I'm personally also using a template without a parent for my sites. (More precisely I have my own base template, and some templates deriving from my base.)